Portability fixes:

- Do not assume that if !__GLIBC__ lseek() cannot handle 64-bit offsets.

--- grub-core/kern/emu/hostdisk.c.orig	2012-06-25 10:32:04.000000000 +0200
+++ grub-core/kern/emu/hostdisk.c	2014-12-28 09:06:09.224044788 +0100
@@ -44,8 +44,7 @@
 #ifdef __linux__
 # include <sys/ioctl.h>         /* ioctl */
 # include <sys/mount.h>
-# if !defined(__GLIBC__) || \
-        ((__GLIBC__ < 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ < 1)))
+# if ((__GLIBC__ < 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ < 1)))
 /* Maybe libc doesn't have large file support.  */
 #  include <linux/unistd.h>     /* _llseek */
 # endif /* (GLIBC < 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR < 1)) */
@@ -761,8 +760,7 @@ linux_find_partition (char *dev, grub_di
 }
 #endif /* __linux__ */
 
-#if defined(__linux__) && (!defined(__GLIBC__) || \
-        ((__GLIBC__ < 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ < 1))))
+#if (defined(__linux__) && defined(__GLIBC__)) && (((__GLIBC__ < 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ < 1))))
   /* Maybe libc doesn't have large file support.  */
 grub_err_t
 grub_util_fd_seek (int fd, const char *name, grub_uint64_t off)
